File: src/main/java/net/dankrushen/danknn/danklayers/DankOutputLayer.java
package net.dankrushen.danknn.danklayers;

import net.dankrushen.danknn.DankConnection;
import net.dankrushen.danknn.DankNeuron;

import java.util.ArrayList;

public class DankOutputLayer extends DankLayer implements IDankOutputLayer {
    private IDankInputLayer inputLayer;
    private DankConnection[] inConnections;

    public DankOutputLayer(int numNeurons, IDankInputLayer inputLayer) {
        super(numNeurons);

        this.inputLayer = inputLayer;

        inputLayer.setOutputLayer(this);

        ArrayList<DankConnection> inConnectionsList = new ArrayList<>();

        for (DankNeuron externalNeuron : inputLayer.getNeurons()) {
            for (DankNeuron internalNeuron : getNeurons()) {
                inConnectionsList.add(new DankConnection(externalNeuron, internalNeuron));
            }
        }

        inConnections = inConnectionsList.toArray(new DankConnection[]{});
    }

    @Override
    public IDankInputLayer getInputLayer() {
        return inputLayer;
    }

    @Override
    public DankConnection[] getInputConnections() {
        return inConnections;
    }
}

File: src/main/java/net/dankrushen/danknn/danklayers/DankOutputLayer.kt
package net.dankrushen.danknn.danklayers

import net.dankrushen.danknn.DankConnection
import java.util.*

class DankOutputLayer(numNeurons: Int, override val inputLayer: IDankInputLayer) : DankLayer(numNeurons), IDankOutputLayer {
    override val inputConnections: Array<DankConnection>

    init {

        inputLayer.outputLayer = this

        val inConnectionsList = ArrayList<DankConnection>()

        for (externalNeuron in inputLayer.neurons) {
            for (internalNeuron in neurons) {
                inConnectionsList.add(DankConnection(externalNeuron, internalNeuron))
            }
        }

        inputConnections = inConnectionsList.toTypedArray()
    }
}
